bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/internal/AudioServlet.java

```diff
@@ -127,32 +140,50 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
             }
 
             ServletOutputStream os = resp.getOutputStream();
-            IOUtils.copy(stream, os);
+
+            InputStream is = null;
+            if (multiAccess) {
+                // we need to care about concurrent access and have a separate stream for each thread
+                try {
+                    is = ((FixedLengthAudioStream) stream).getClonedStream();
+                } catch (AudioException e) {
+                    resp.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getMessage());
+                }
+            } else {
+                is = stream;
+            }
+            IOUtils.copy(is, os);
             resp.flushBuffer();
             IOUtils.closeQuietly(stream);
         }
     }
 
+    private synchronized void removeTimedOutStreams() {
+        for (String streamId : multiTimeStreams.keySet()) {
+            if (streamTimeouts.get(streamId) < System.currentTimeMillis()) {
+                // the stream has expired, we need to remove it!
+                FixedLengthAudioStream stream = multiTimeStreams.remove(streamId);
+                streamTimeouts.remove(streamId);
+                IOUtils.closeQuietly(stream);
+                stream = null;
+                logger.debug("Removed timed out stream {}", streamId);
+            }
+        }
+    }
+
     @Override
     public URL serve(AudioStream stream) {
         String streamId = UUID.randomUUID().toString();
-        streams.put(streamId, stream);
+        oneTimeStreams.put(streamId, stream);
         return getURL(streamId);
     }
 
     @Override
-    public URL serveWithSize(AudioStream stream) {
-        if (stream instanceof FixedLengthAudioStream) {
-            return serve(stream);
-        } else if (stream.markSupported()) {
-            DiscreteAudioStream streamWithSize = new DiscreteAudioStream(stream);
-            return serve(streamWithSize);
-        } else {
-            // TODO: We should also support streams without mark support, but this will mean that we have to read it
-            // first to memory or file system and create a new AudioStream from there.
-            logger.warn("Stream cannot be reset, so it is served without size through HTTP");
-            return serve(stream);
-        }
+    public URL serve(FixedLengthAudioStream stream, int seconds) {
+        String streamId = UUID.randomUUID().toString();
+        multiTimeStreams.put(streamId, stream);
+        streamTimeouts.put(streamId, System.currentTimeMillis() + (seconds * 1000L));
```

Comment:
I would prefer to use `System.nanoTime()` for such time, as it is independent of the system time and cannot be corrupted by time changes.